core/src/test/java/org/apache/accumulo/core/spi/compaction/DefaultCompactionPlannerTest.java

```diff
@@ -239,6 +243,38 @@ public void testMaxSize() {
     assertEquals(CompactionExecutorIdImpl.externalId("large"), job.getExecutor());
   }
 
+  /**
+   * Tests max.open can be overridden with maxOpen.
+   */
+  @Test
+  @SuppressWarnings("removal")
+  public void testOverrideMaxOpen() {
+    DefaultCompactionPlanner planner = new DefaultCompactionPlanner();
+    ConfigurationCopy aconf = new ConfigurationCopy(DefaultConfiguration.getInstance());
+    aconf.set(Property.TSERV_MAJC_THREAD_MAXOPEN.getKey(), "5");
+
+    ConfigurationImpl config = new ConfigurationImpl(aconf);
+
+    ServiceEnvironment senv = EasyMock.createMock(ServiceEnvironment.class);
+    EasyMock.expect(senv.getConfiguration()).andReturn(config).anyTimes();
+    EasyMock.replay(senv);
+
+    String executors = getExecutors("'type': 'internal','maxSize':'32M','numThreads':1",
+        "'type': 'internal','maxSize':'128M','numThreads':2", "'type': 'external','queue':'q1'");
+
+    // Test with max.open property
+    planner.init(getInitParams(senv, executors, "15"));
+    assertEquals(5, planner.getMaxFilesToCompact());
+
+    // Test again by overriding maxOpen
+    Integer maxOpen = 30;
+    aconf.set(Property.TSERV_COMPACTION_SERVICE_PREFIX.getKey() + "cs1.planner.opts.maxOpen",
+        maxOpen.toString());
+    planner = new DefaultCompactionPlanner();
+    planner.init(getInitParams(senv, executors, maxOpen.toString()));
+    assertEquals(maxOpen, planner.getMaxFilesToCompact());
```

Comment:
Could set max open to 10 and pass in 11 files of all the same size  like the prev comment and verify it creates a job of size 10.